### PR TITLE
RTI-3237 full row edit batch bug

### PR DESCRIPTION
**Pull request by SalvatorePreviti**
fix RTI-3237
fix RTI-3230

## Bug Fixes

-   **Full-row batch edit styling lost on row navigation** — Fixed a bug where the `ag-cell-batch-edit` and `ag-row-batch-edit` CSS classes were removed from previously edited cells/rows when navigating to a different row during full-row batch editing. The stop/cancel flow now distinguishes between "stop editing this row to move to the next" (preserves batch pending values) and "commit the entire batch" via a new `commit` flag threaded through `StopEditParams` and the edit strategies.

-   **Fill handle reduction cleared cell data instead of restoring source values** — When shrinking a fill handle selection during batch editing, affected cells had their data cleared instead of being restored to their original source values. The fill handle reduction path now uses `editModelSvc.removeEdits()` to properly restore cells to their pre-batch source values.

-   **Escape in batch mode reverted to source data instead of previous pending value** — When re-editing a cell that already had a batch pending value, pressing Escape was reverting to the original source data value rather than the previous batch pending value. The batch Escape path now skips persisting editor content and calls `model.stop()` with `preserveBatch: true`, so the prior pending value survives.

-   **`batchEditingStarted` and `batchEditingStopped` inconsistently raised, with no re-entrancy, and sometime raised more than once. Added guards and a mechanism to ensure they are always raised at the right time and only if an edit actually happens.

-   **`batchEditingStopped` skipped when all edits were purged** — If a user typed a value then reverted it (so all edits were purged), `prepareStopContext` would return `null` and `dispatchBatchStopped` was never reached, leaving `batchEditingStarted` unbalanced. `stopBatchEditing` now fires a balancing `batchEditingStopped` with cancel semantics if `batchStartDispatched` is still true after `stopEditing` returns.

-   **`_syncFromEditor` sourceValue check used falsy instead of `undefined`** — The condition `!edit?.sourceValue` was changed to `edit?.sourceValue === undefined`. Previously, a falsy source value like `0`, `''`, or `false` would incorrectly trigger re-initialisation of the source value.

-   **Stale CellCtrl reference after `setDataValue`** — In `editService.setDataValue()`, the CellCtrl is now re-fetched after `setNodeDataValue` instead of reusing a previously captured reference, because change detection during `setDataValue` may have recreated the CellCtrl.

-   **`rowValueChanged` no longer fires prematurely during batch row-to-row Tab navigation** — During full-row batch editing, tabbing across row boundaries was dispatching `rowValueChanged` even though values were only staged, not committed. `rowValueChanged` now only fires when `commit` is true or batch editing is not active. `commitBatchEdit()` now passes `commit: true` to `stopEditing()` so the event fires correctly on final commit.

-   **`committing` flag not reset on exception** — `setNodeDataValue` and `applyBulkEdit` now wrap `committing = true` in try/finally blocks so the flag is always cleared even if `rowNode.setDataValue()` or `stopEditing()` throws.

-   **`bulkEditingStopped` dispatched without prior `bulkEditingStarted`** — In `applyBulkEdit`, `bulkEditingStopped` was dispatched unconditionally in the non-batch path, even though `bulkEditingStarted` is only dispatched when `!this.batch`. Added a `bulkStartDispatched` local flag to track whether the start event was fired, and gate the stop dispatch accordingly. The stop dispatch is placed inside `finally` so it always balances the start event even on exception.

## Internal Changes

-   **`editModelService.stop()` signature now requires explicit args** — Changed from optional `position?` with fallback to `this.clear()` to mandatory `(position, preserveBatch, cancel)`.

-   **`restoreSourceInBatch` for fill-handle reduction** — Fill handle reduction now uses `editModelSvc.removeEdits()` to restore cells to their original source values instead of clearing them.

-   **`ensureBatchStarted()` utility** — Lazily dispatches `batchEditingStarted` on the first write or editor open during a batch session. Placed consistently after model mutation at all write paths.

-   **`dispatchBatchStopped()` utility** — Extracted from `finishStopEditing`; handles filtering edits on commit, safety `ensureBatchStarted()` when real edits exist, and conditionally dispatching `batchEditingStopped`.

## Refactoring

-   Split `BaseEditStrategy.stop()` into `stopCancelled()` and `stopCommitted()` for clearer separation of cancel vs commit flows.
-   Added `syncEditAfterCommit()` helper in `EditService` to keep `sourceValue` in sync after non-batch commits (handles custom valueSetter transforms).
-   `_destroyEditor` now preserves prior edit state for cells with a previous editor session (`cellStoppedEditing`).
-   Extracted `forEachEditableCellInRanges()` helper in `RangeService` for common iteration logic.
-   Split `editService.setDataValue()` into `applyDirectValue()` (editApi/undoRedoApi path) and `applyExistingEdit()` (existing edit reconciliation) to reduce cognitive complexity.
-   Replaced `Object.assign` with spread syntax and `?? false` with `!!` for consistency.
-   Made `StopContext` fields non-optional with explicit defaults for type safety.

## Test Coverage

New tests covering batch styling, clear, range paste, full-row batch, batch clear, and edge cases including:
-   `setDataValue` return value correctness with `cellClear` source in batch mode.
-   `batchEditingStarted`/`batchEditingStopped` events for paste-only, cellClear-only, and no-change batch sessions.
-   Delete range in batch mode fires batch events correctly.

### CODE REVIEW

**Review by StephenCooper — APPROVED** [review]

File: packages/ag-grid-enterprise/src/rangeSelection/rangeService.ts [hunk 1]
```diff
@@ -847,22 +850,17 @@ export class RangeService extends BeanStub implements NamedBean, IRangeService,
...
+        const { valueSvc, editSvc } = beans;
+        const batch = !!editSvc?.isBatchEditing();
+
+        this.forEachEditableCellInRanges(cellRanges, (rowNode, column) => {
+            if (restoreSourceInBatch && batch) {
+                editSvc?.batchResetToSourceValue({ rowNode, column });
+                return;
+            }
+            const deleteValue = valueSvc.getDeleteValue(column, rowNode);
+            rowNode.setDataValue(column, deleteValue, cellEventSource);
```
[the remark this answers is not in the document]
Comment (SalvatorePreviti) [reply packages/ag-grid-enterprise/src/rangeSelection/rangeService.ts:862]: restoreSourceInBatch does exactly this - it removes the edit without calliung setDataValue - false positive

**Review by gportela85 — APPROVED** [review]
